src/ort_genai_c.h

```diff
@@ -492,6 +492,22 @@ OGA_EXPORT OgaResult* OGA_API_CALL OgaProcessorProcessImagesAndAudios(const OgaM
 OGA_EXPORT OgaResult* OGA_API_CALL OgaTokenizerDecode(const OgaTokenizer*, const int32_t* tokens, size_t token_count, const char** out_string);
 OGA_EXPORT OgaResult* OGA_API_CALL OgaProcessorDecode(const OgaMultiModalProcessor*, const int32_t* tokens, size_t token_count, const char** out_string);
 
+/**
+ * @brief Applies a chat template to the given input.
+ *
+ * This function processes the specified template with the provided input using the
+ * tokenizer, and outputs the resulting string. Optionally, it can include a
+ * generation prompt in the output.
+ *
+ * \param[in] tokenizer OgaTokenizer used for template processing.
+ * \param[in] template_str Null-terminated string representing the chat template
```

Comment:
Should the user be responsible to pass in the chat template? Can it read from the tokenizer config or from chat template config? Is this an optional input that overrides what is found in the tokenizer config?

Reply:
In most cases, the user does not know the chat template and would rather prefer we auto set it from the tokenizer config. Having said that, there could be some scenarios where the user does want to override the template.

But it should not be a required input.

Reply:
ok, can we mention it here that it is optional? It wasn't clear to me that this was an optional input.
Also, we probably should avoid adding c api utilities just for testing if possible.

Would users want to override this template str?

Reply:
it's mentioned in the brief:
> Optionally, it can include a generation prompt in the output.

https://github.com/microsoft/onnxruntime-genai/pull/1398/files#diff-4d2efedaefd309b86072862c893606033ca895483d661fb936541bc1774ca3f9R499

Reply:
Sorry, I meant can we document the template str as optional? The documentation doesn't do that at the moment.

Reply:
the reason i want to avoid that is because it cannot just be left in this case; say with the LITE api Randy made for instance we could actually have optional params, but in this case, if it's not being used, the user of this API needs to pass in a nullptr like this: https://github.com/microsoft/onnxruntime-extensions/blob/d944f38acec75a4ff73dfe40ad4c082e7bb33c53/test/pp_api_test/test_tokenizer_chat.cc#L39

Reply:
```suggestion
 * \param[in] template_str Null-terminated string representing the chat template. Use nullptr to fall back to the default chat template from the tokenizer config.
```

Reply:
gotcha, sounds good, thanks - just committed!